pkg/controllers/user/alert/watcher/cluster_scan.go

```diff
@@ -0,0 +1,161 @@
+	return cs, nil
+}
+
+func (csw *ClusterScanWatcher) Updated(cs *v3.ClusterScan) (runtime.Object, error) {
+	if cs == nil {
+		return nil, nil
+	}
+	if cs.DeletionTimestamp != nil {
+		return cs, nil
+	}
+	// Start with Unknown, True if there is/are a matching alert rule(s), else False
+	if !(v3.ClusterScanConditionAlerted.IsUnknown(cs) &&
+		v3.ClusterScanConditionCompleted.IsTrue(cs)) {
+		return cs, nil
+	}
+	var err error
+	if csw.alertManager.IsDeploy == false {
+		logrus.Infof("ClusterScanWatcher: Sync: alert manager not deployed")
+		v3.ClusterScanConditionAlerted.False(cs)
+		v3.ClusterScanConditionAlerted.Message(cs, MsgAlertManagerNotDeployed)
+		return cs, nil
+	}
+	clusterAlertRules, err := csw.clusterAlertRuleLister.List("", labels.NewSelector())
+	if err != nil {
+		return cs, fmt.Errorf("ClusterScanWatcher: Sync: error listing cluster alert rules: %v", err)
+	}
+
+	match := false
+	var matchingAlertRules []*v3.ClusterAlertRule
+	for _, alertRule := range clusterAlertRules {
+		if alertRule.Status.AlertState == "inactive" || alertRule.Spec.ClusterScanRule == nil {
+			continue
+		}
+		if csw.isAlertRuleMatching(cs, alertRule) {
+			match = true
+			matchingAlertRules = append(matchingAlertRules, alertRule)
+		}
+	}
+	logrus.Debugf("ClusterScanWatcher: Sync: matchingAlertRules: %+v", matchingAlertRules)
+	if !match {
+		v3.ClusterScanConditionAlerted.False(cs)
+		v3.ClusterScanConditionAlerted.Message(cs, MsgNoMatchingAlertRule)
+		return cs, nil
+	}
+
+	alertSuccessful := true
+	for _, alertRule := range matchingAlertRules {
+		if e := csw.sendAlert(cs, alertRule); e != nil {
+			alertSuccessful = false
+			logrus.Errorf("ClusterScanWatcher: Sync: error sending alert: %v", e)
+			err = multierror.Append(err, e)
+		}
+	}
+	if !alertSuccessful {
+		return cs, err
+	}
+	v3.ClusterScanConditionAlerted.True(cs)
+	logrus.Infof("ClusterScanWatcher: Sync: %+v", cs)
+	return cs, nil
+}
+
+func (csw *ClusterScanWatcher) isAlertRuleMatching(cs *v3.ClusterScan, alertRule *v3.ClusterAlertRule) bool {
+	if alertRule.Spec.ClusterScanRule.ScanRunType != cs.Spec.RunType {
+		return false
+	}
+	if alertRule.Spec.ClusterScanRule.FailuresOnly && !(cs.Status.CisScanStatus.Fail > 0) {
+		return false
+	}
+	return true
+}
+
+func (csw *ClusterScanWatcher) sendAlert(cs *v3.ClusterScan, alertRule *v3.ClusterAlertRule) error {
+	ruleID := common.GetRuleID(alertRule.Spec.GroupName, alertRule.Name)
+	clusterDisplayName := common.GetClusterDisplayName(csw.clusterName, csw.clusterLister)
+
+	data := map[string]string{}
+	data["rule_id"] = ruleID
+	data["group_id"] = alertRule.Spec.GroupName
+	data["alert_type"] = constants.AlertTypeClusterScan
+	data["alert_name"] = alertRule.Spec.DisplayName
+	data["severity"] = alertRule.Spec.Severity
+	data["cluster_name"] = clusterDisplayName
+	data["component_name"] = cs.Name
+	data["logs"] = csw.getAlertMessage(cs, alertRule)
+
+	logrus.Infof("ClusterScanWatcher: sendAlert: about to send alert")
```

Comment:
nit: needs to have cluster+alert name

Reply:
actually, I had them while debugging, will remove the logs.